homework37/main.js

```diff
@@ -0,0 +1,116 @@
+import { isRequired } from "./modules/validation.js";
+import {showError, showSuccess, removeValidClass } from "./modules/formUI.js";
+
+const api_Endpoint = 'https://jsonplaceholder.typicode.com';
+const form = document.forms.getPostById;
+const postsInfo = document.querySelector(".posts__info");
+const commentSection = document.querySelector(".comments-section");
+const showComments = document.querySelector(".showComments");
+const hideComments = document.querySelector(".hideComments");
+
+const getIdNumInput = () => document.getElementById("idNum");
+
+const getIdNumValue = () => getIdNumInput().value.trim();
+
+const renderCartInfo = (data) => {
+    postsInfo.style.display = "block";
+
+    const titleOfPost = postsInfo.querySelector(".card-title");
+    const numOfPost = postsInfo.querySelector(".idVal");
+    const postText = postsInfo.querySelector(".card-text");
+
+    titleOfPost.innerHTML = `${data.title}`;
+    numOfPost.innerHTML = `${data.id}`;
+    postText.innerHTML = `${data.body}`;
+};
+
+const renderComments = (data) => {
+    data.forEach(element => {
+        commentSection.insertAdjacentHTML("beforeend", `
+        <h5 class="card-title">Комментар від ${element.name}</h5>
+        <h6 class="card-subtitle">${element.email}</h6>
+        <p class="card-text">${element.body}</p>`);
+    });
+};
```

Comment:
```
const renderComments = (data) => {
  data.forEach(item => {
    const content = (`
      <h5 class="card-title">
      Комментар від ${item.name}
      </h5>
      <h6 class="card-subtitle">
        ${item.email}
      </h6>
      <p class="card-text">
        ${item.body}
      </p>
    `);

    // Not so good
    // Call insertAdjacentHTML N times because inside loop
    commentSection.insertAdjacentHTML("beforeend", content);
  });
};

const renderComments = (data) => {
  const content = data.map(item => (`
    <h5 class="card-title">
    Комментар від ${item.name}
    </h5>
    <h6 class="card-subtitle">
      ${item.email}
    </h6>
    <p class="card-text">
      ${item.body}
    </p>
  `)).join("");

  // Much better
  // Call insertAdjacentHTML once because outside the loop
  commentSection.insertAdjacentHTML("beforeend", content);
};

const renderComments = (data) => {
  const content = data.reduce((html, item) => html + (`
    <h5 class="card-title">
    Комментар від ${item.name}
    </h5>
    <h6 class="card-subtitle">
      ${item.email}
    </h6>
    <p class="card-text">
      ${item.body}
    </p>
  `), "");

  // Much better
  // Call insertAdjacentHTML once because outside the loop
  commentSection.insertAdjacentHTML("beforeend", content);
};
```